lib/plugin/selenoid.js

```diff
@@ -341,6 +342,48 @@ function setOptionsForWebdriver(config) {
   }));
 }
 
+function setOptionsForPuppeteer(config) {
+  const { remote } = require('webdriverio');
+  const Puppeteer = container.helpers('Puppeteer');
+  let browser = null;
+
+  const lauchPuppeteer = (test) => {
+    console.log('before lauch');
+    recorder.add('creating browser session', async () => {
+      console.log('Starting browser session');
+      browser = await remote({
+        hostname: 'localhost',
+        capabilities: {
+          browserName: 'chrome',
+          browserVersion: 'latest',
+          // @ts-ignore
+          'selenoid:options': {
+            ...config,
+            name: test.title,
+            videoName: `${test.title}.mp4`,
+            logName: `${test.title}.log`,
+          },
+        },
+      });
+      console.log(browser.sessionId);
+      Puppeteer.isRemoteBrowser = true;
+      Puppeteer.puppeteerOptions = { browserWSEndpoint: `ws://localhost:4444/devtools/${browser.sessionId}/` };
+    });
+  };
+
+  // Launch browser for every test before
+
+  event.dispatcher.on(event.test.before, lauchPuppeteer);
```

Comment:
We must have parameter to restart chrome.
Because on each test restarting chrome is not advisable.